formats/json/commonMain/src/kotlinx/serialization/json/internal/lexer/AbstractJsonLexer.kt

```diff
@@ -759,4 +748,37 @@ internal abstract class AbstractJsonLexer {
             currentPosition = snapshot
         }
     }
+
+    fun consumeCommaOrPeekEnd(
+        allowTrailing: Boolean,
+        expectedEnd: Char,
+        entity: String = "object",
+    ) {
+        val nextToken = peekNextToken()
+        if (nextToken == charToTokenClass(expectedEnd)) return
+        if (nextToken != TC_COMMA) {
+            path.popDescriptor()
+            fail("Expected end of the $entity or comma")
+        }
+        val commaPosition = currentPosition++ // consumes peeked comma
```

Comment:
I was caught off-guard by this, since my "intuition" told me to write code like 
```kotlin
val peeked = peekNextToken()
check(peeked)
// consume peeked token
consumeNextToken()
```
but for some reason, implementation of `consumeNextToken` DOES NOT look at `currentPosition` and immediately starts from `currentPosition++`, ignoring "peeked" token

Reply:
That's strange. You can see here that `consumeNextToken` starts from `currentPosition`: https://github.com/Kotlin/kotlinx.serialization/blob/1ffdbafe2f417846d4998669c6943e61b5ee82eb/formats/json/commonMain/src/kotlinx/serialization/json/internal/lexer/StringJsonLexer.kt#L20

And than advances it (`cpos++`). So currentPosition points to the first unconsumed character, and `consumeNextToken` always advances it at least by one. But it first will return the char at `currentPosition`, unless it is not a whitespace.

Reply:
oops, I am sorry, you are right, I mistook other error for this. Fixed it now, thanks!